scripts/zones/Attohwa_Chasm/npcs/Jakaka.lua

```diff
@@ -40,5 +63,18 @@ end;
 function onEventFinish(player,csid,option)
 	-- printf("CSID: %u",csid);
 	-- printf("RESULT: %u",option);
+    if(csid == 12) then
+        local day = 86400; -- Number of seconds in a day
+		player:addKeyItem(MIASMA_FILTER);
+		player:messageSpecial(KEYITEM_OBTAINED,MIASMA_FILTER);
+		player:setVar("[ENM]MiasmaFilter",os.time(t)+86400*MIASMA_FILTER_COOLDOWN); -- Current time + 1day*MIASMA_FILTER_COOLDOWN
```

Comment:
@xdemolish Unless multiple NPC's would use a var, why define it in a global instead of in file? It can be done in file with the exact same descriptive name so human readability isn't an issue.